src/Adapter/Product/Update/ProductShopUpdater.php

```diff
@@ -139,4 +148,24 @@ private function copyCarriersToShop(Product $sourceProduct, ShopId $targetShopId
             ShopConstraint::shop($targetShopId->getValue())
         );
     }
+
+    /**
+     * @param ProductId $productId
+     * @param ShopId $sourceShopId
+     * @param ShopId $targetShopId
+     *
+     * @return void
+     *
+     * @throws \PrestaShop\PrestaShop\Core\Domain\Shop\Exception\ShopException
+     * @throws \PrestaShop\PrestaShop\Core\Exception\CoreException
+     */
+    public function copyImageAssociations(ProductId $productId, ShopId $sourceShopId, ShopId $targetShopId): void
+    {
+        //todo: update test
+        $imagesFromSourceShop = $this->productImageMultiShopRepository->getImages($productId, ShopConstraint::shop($sourceShopId->getValue()));
+        foreach ($imagesFromSourceShop as $image) {
+            $image->id_product = $productId->getValue();
+            $image->associateTo($targetShopId->getValue());
```

Comment:
Here again you need to specify the `$image->id_product` so there is definitely something wrong here, probably related to the ObjectModel but anyway it's worrying Not blocking for this PR but it should be investigated